tt_metal/llrt/tt_memory.cpp

```diff
@@ -24,10 +24,13 @@ memory::memory() {
     packed_size_ = 0;
 }
 
-memory::memory(std::string const &path) : memory() {
+memory::memory(std::string const &path, Relocate relo_type) : memory() {
     ElfFile elf;
 
     elf.ReadImage(path);
+    if (relo_type == Relocate::XIP) {
+        elf.MakeExecuteInPlace();
```

Comment:
When copying the segment info below, you're not checking segment.relocs, and it would be good to assert that is `empty()`, so it blows up if ever there are dynamic relocs than need to be handled.

Reply:
I dont't follow and I don't know the elf file object - can you paste code that I should throw in here?

Reply:
the xipification patch adds a `std::vector<offset_t>` to the Segment class. It adds locations that need contain a R_RISCV_32 reloc, which need a dynamic adjustment.  So need to assert that vector is empty at the moment:
```
   auto emit_segment = [&](ElfFile::Segment const& segment) {
        assert(relocs_.empty() && "dynamic relocations present"); <-- this line
	link_spans_.emplace_back(
            segment.address, segment.contents.size());
	data_.insert(data_.end(), segment.contents.begin(), segment.contents.end());
	total_size += segment.contents.size();
    };
```